robottelo/host_helpers/api_factory.py

```diff
@@ -711,6 +713,217 @@ def wait_for_errata_applicability_task(
+
+        Attempt to make needed associations between detached entities.
+
+        Add desired repos to the content-view prior to calling this helper.
+        Or, add them to content-view after calling, then publish/promote.
+
+        param satellite: sat instance where needed entities exist.
+        param client: instance of a rhel contenthost to register.
+        param enable_repos (bool): enable all available repos on the client, after registration?
+            default: False, be sure to enable any repo(s) for client, after calling this method.
+        param rex_key (bool): add a Remote Execution Key to client for satellite?
+            default: False
+        param force (bool): force registration of the client to bypass?
+            default: False
+
+        type: Below arguments can be any of the following:
+            - int: pass id of entity to be read
+            - str: pass name of entity to be searched
+            - entity: pass an entity instance
+        param organization: pass an Organization instance, name, or id to use.
+        param activation_key: pass an Activation-Key instance, name, or id.
+        param environment: pass a Lifecycle-Environment instance, name, or id.
+            for example: can pass string name 'Library'.
+        param content_view: pass a Content-View instance, name, or id.
+            for example: can pass string name 'Default Organization View'.
+
+        Notes:
+            1. Will fail if passed entities do not exist in the same organization and satellite.
+            2. You can use param `enable_repos`, to try enabling any repositories on client,
+                that were added to content-view prior. But if there are no
+                repositories added/made available, this will fail.
+            3. The Default Organization View cannot be published, promoted, edited, or deleted.
+                but you can register the client to it.
+
+        Steps:
+            1. get needed entities from arguments, id, name, or instance. Read all as instance.
+            2. publish the content-view if no versions exist, or needs_publish.
+            3. promote the newest content-view-version if not in environment already. Skip for 'Library'.
+            4. assign environment and content-view to the activation-key, if not associated.
+            5. Register the host, using the activation-key associated with content.
+            6. Add a rex_key to the client if desired, enable all available repositories if desired.
+
+        return: dict containing the updated entities:
+            {
+                'client', 'organization', 'activation_key',
+                'environment', 'content_view',
+            }
+        """
+        entities = {
+            'Organization': organization,
+            'ActivationKey': activation_key,
+            'LifecycleEnvironment': environment,
+            'ContentView': content_view,
+        }
+        assert hasattr(client, 'hostname')
```

Comment:
This is a helper method, not a fixture or test, probably best to leave out assertions and just handle the case where this fails more appropriately like a return condition.